### PR TITLE
Enforce minimum version

### PR DESCRIPTION
**Pull request by nwithan8**
EasyPost explicitly supports Java 8+ in this client library (compatibility with earlier versions is untested and not guaranteed)

- GitHub Action checks code compatibility in Java 8 - latest (Java 17)
- Maven plugin restricts code development to Java 8 - latest (Java 17)
- Maven CheckStyle plugin will fail on anything older than Java 8, so development needs to be done in Java 8+ environment

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -182,6 +189,26 @@
...
+                <executions>
+                    <execution>
+                        <id>enforce-java</id>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireJavaVersion>
+                                    <version>[1.6.0,)</version>
```
Comment (jchen293) [inline pom.xml:205]: Is this a mismatching bracket?

Comment (jchen293) [reply pom.xml:205]: Nvm, I think this is correct.

Comment (Justintime50) [reply pom.xml]: Yes, this is correct based on this: https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html

**Review by Justintime50 — APPROVED** [review]
Nice work!

**Review by jchen293 — APPROVED** [review]
lgtm

**Review by Justintime50 — DISMISSED** [review]

**Review by jchen293 — APPROVED** [review]
lgtm

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]
lgtm

**Pull request merged**